src/ostorlab/serve_app/types.py

```diff
@@ -342,3 +342,55 @@ class OxoScansType(graphene.ObjectType):
     """Graphene object type for a list of scans."""
 
     scans = graphene.List(OxoScanType, required=True)
+
+
+class AgentArgumentInputType(graphene.InputObjectType):
+    """Input object type for an agent argument."""
+
+    name = graphene.String(required=True)
+    type = graphene.String(required=True)
+    description = graphene.String(required=False)
+    value = common.Bytes(required=False)
+
+
+class AgentGroupAgentCreateInputType(graphene.InputObjectType):
+    """Input object type for creating an agent group agent."""
+
+    agent_key = graphene.String(required=True)
+    args = graphene.List(AgentArgumentInputType)
+
+
+class AgentGroupCreateInputType(graphene.InputObjectType):
+    """Input object type for creating an agent group."""
+
+    name = graphene.String(required=True)
+    description = graphene.String(required=True)
+    agents = graphene.List(AgentGroupAgentCreateInputType, required=True)
+
+
+class AgentGroupType(graphene_sqlalchemy.SQLAlchemyObjectType):
+    """SQLAlchemy object type for an agent group."""
+
+    key = graphene.String()
+
+    class Meta:
+        model = models.AgentGroup
+
+        only_fields = (
+            "id",
+            "name",
+            "description",
+            "created_time",
+        )
+
+    def resolve_key(self: models.AgentGroup, info: graphql_base.ResolveInfo) -> str:
```

Comment:
Shouldn't we also resolve the list of agents or no need to?

Reply:
@3asm shall we ? because we dont do that in RE spec